docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Contents
+1. [Introduction](#introduction)
+    1. [Background](#background)
+    1. [Purpose](#purpose)
+    1. [Scope](#scope)
```

Comment:
Similar to what was mentioned below, it feel it would be better to number the sub sections in the 1.1, 1.2 format as well.